presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1228,6 +1217,38 @@ private ActualProperties derivePropertiesRecursively(PlanNode result)
         {
             return PropertyDerivations.derivePropertiesRecursively(result, metadata, session, types, parser);
         }
+
+        private ExchangeNode createPartitionedRemoteExchange(PlanNode child, List<Symbol> partitioningColumns, Optional<Symbol> hashColumn)
+        {
+            return createPartitionedRemoteExchange(child, partitioningColumns, hashColumn, false);
+        }
+
+        private ExchangeNode createPartitionedRemoteExchange(PlanNode child, List<Symbol> partitioningColumns, Optional<Symbol> hashColumn, boolean replicateNullsAndAny)
```

Comment:
An alternative way would be to add an overload to `ExchangeNode.partitionedExchange` that also takes `Partitioning` (instead of kind of copy `partitionedExchange` to be `createPartitionedRemoteExchange`). What do you think?